radix-engine/src/model/substates/substate.rs

```diff
@@ -20,23 +18,116 @@ pub enum Substate {
     KeyValueStoreEntry(KeyValueStoreEntrySubstate),
 }
 
-impl Substate {
+impl Into<VaultSubstate> for PersistedSubstate {
+    fn into(self) -> VaultSubstate {
+        if let PersistedSubstate::Vault(vault) = self {
+            vault
+        } else {
+            panic!("Not a vault");
+        }
+    }
+}
```

Comment:
Why do we need to add conversion for `VaultSubstate` only?

Reply:
Currently used for `test_runner` to inspect the vault without seeing the runtime representation. All other substates just use `to_runtime().into()` since they have the same runtime/persisted representations.